src/main/noise/Velvet.cpp

```diff
@@ -0,0 +1,331 @@
+        {
+            return sRandomizer.random(RND_LINEAR);
+        }
+
+        float Velvet::get_spike()
+        {
+            switch (enCore)
+            {
+                case VN_CORE_MLS:
+                    return sMLS.process_single(); // Either 1 or -1
+                default:
+                case VN_CORE_LCG:
+                    return 2.0f * roundf(get_random_value()) - 1.0f; // Either 1 or -1
+            }
+        }
+
+        float Velvet::get_crushed_spike()
+        {
+            float spike = 0.0f;
+
+            if (get_random_value() > sCrushParams.fCrushProb)
+                spike = 1.0f;
+
+            return 2.0f * spike - 1.0f; // Either 1 or -1
+        }
+
+        void Velvet::do_process(float *dst, size_t count)
+        {
+
+            switch (enVelvetType)
+            {
+
+                case VN_VELVET_OVN:
+                {
+                    dsp::fill_zero(dst, count);
+
+                    size_t idx = 0;
+                    size_t scan = 0;
+
+                    while (idx < count)
+                    {
+                        idx = scan * fWindowWidth + get_random_value() * (fWindowWidth - 1.0f);
+
+                        if (sCrushParams.bCrush)
+                            dst[idx] = get_crushed_spike();
+                        else
+                            dst[idx] = get_spike();
+
+                        ++scan;
+                    }
+                }
+                break;
+
+                case VN_VELVET_OVNA:
+                {
+                    dsp::fill_zero(dst, count);
+
+                    size_t idx = 0;
+                    size_t scan = 0;
+
+                    while (idx < count)
+                    {
+                        idx = scan * fWindowWidth + get_random_value() * fWindowWidth;
+
+                        if (sCrushParams.bCrush)
+                            dst[idx] = get_crushed_spike();
+                        else
+                            dst[idx] = get_spike();
+
+                        ++scan;
+                    }
+                }
+                break;
+
+                case VN_VELVET_ARN:
+                {
+                    dsp::fill_zero(dst, count);
+
+                    size_t idx = 0;
+
+                    while (idx < count)
+                    {
+                        idx += 1.0f + (1.0f - fARNdelta) * (fWindowWidth - 1.0f) + 2.0f * fARNdelta * (fWindowWidth - 1.0f) * get_random_value();
+
+                        if (sCrushParams.bCrush)
+                            dst[idx] = get_crushed_spike();
+                        else
+                            dst[idx] = get_spike();
+                    }
+                }
+                break;
+
+                case VN_VELVET_TRN:
+                {
+                    while (count--)
+                    {
+                        float value = roundf(fWindowWidth * (get_random_value() - 0.5f) / (fWindowWidth - 1.0f));
```

Comment:
I would like to replace division with multiplication moving the single divisin outside the `while` loop:
```c++
float k = fWindowWidth / (fWindowWidth - 1.0f);
while (count--)
{
    float value = roundf(k * (get_random_value() - 0.5f))
```